### DISCUSSION TITLE
Understanding what tripped the alarm

### DISCUSSION THREAD

**Discussion by richardcastle** [vanackej/risco-mqtt-local#58] [Q&A]
I'm loving this integration for Home Assistant. It worked straight out of the box using the new ARM build for my RPI4. Very pleased. Kudos to the developers!

As I untangle myself from the Risco Cloud setup, I do miss understanding which sensor tripped the alarm which is part of the alert messaging on the Risco app on my phone. We've got rather large cats in the house and they are normally the culprits.

My question is whether it's possible to create an entity that recalls the sensors that were tripped at the point of the alarm being triggered? I can see that the MQTT stream includes sensor states which accurately report back, so there may be a more involved way of doing this within a HA script, but it would be nice if it came as part of this add-on.

Anyone else found a way to get this information that they want to share?

Thanks.
Richard

**Comment by markxroberts** [comment]
Risco support is a native integration within HA now.  I think this includes binary sensors for triggered alarm sensors.  The latest version of risco-mqtt-local in my repository of the same name (2023.7.1) also includes these features.  I'm not that familiar with GitHub actions and have tried to reproduce creating HA add-ons of the same but have so far had little success, so I can't currently offer you these.  If you're familiar with containers, then it's available through ghcr.io

**Comment by richardcastle** [comment]
Thanks for the response. I did some more research and it appears that other users have approached the problem by looping through all binary sensors and filtering on those that have Risco-specific attributes.

So on an alarm trigger event, a message gets sent with a body containing the warning message and the name of the triggering sensor. I've enclose an example below for anyone who wants to do this for themselves. It seems to work:  

```
message: >
  House alarm triggered! 
    {% for state in states.binary_sensor -%}
      {% if state.state == 'on' and state.attributes.device_class == 'motion' and state.attributes.techLabel == 'Wireless Zone'%}
        {{ state.name }}
      {%- endif -%}
    {%- endfor -%}
```

**Comment by markxroberts** [comment]
Yes, you can do that.  I've had a go at add-on package production.  Not sure whether it's worked!  Config may be slightly different to @vanackej so beware of that if you try it.
